src/sentry/receivers/owners.py

```diff
@@ -0,0 +1,80 @@
+from django.db.models.signals import pre_delete, pre_save
+
+from sentry.models.organization import Organization
+from sentry.models.organizationmember import OrganizationMember
+from sentry.models.team import Team
+from sentry.roles import organization_roles
+
+
+def _assert_org_has_owner_not_from_team(organization, top_role):
+    assert organization.member_set.filter(
+        role=top_role
+    ).exists(), "An organization must have at least one owner"
+
+
+def prevent_demoting_last_owner(instance: OrganizationMember, **kwargs):
+    # if a member is being created
+    if instance.id is None:
+        return
+
+    try:
+        member = OrganizationMember.objects.get(id=instance.id)
+    except OrganizationMember.DoesNotExist:
+        return
+
+    # member is the last owner and the update will remove the last owner
+    assert not (
```

Comment:
AssertionErrors don't look nice to API users. I think a ValidationError would be better.